charts/opentelemetry-collector/values.schema.json

```diff
@@ -699,6 +711,34 @@
         "enabled"
       ]
     },
+    "rollout": {
+      "type": "object",
+      "properties": {
+        "rollingUpdate": {
+          "type": "object",
+          "properties": {
+            "maxSurge": {
+              "$ref": "#/$defs/intOrString"
+            },
+            "maxUnavailable": {
+              "$ref": "#/$defs/intOrString"
+            }
+          }
+        },
+        "strategy": {
+          "type": "string",
+          "enum": [
```

Comment:
It's worth calling out for the review that not all of these strategies are supported for every deployment mode.

`mode: deployment`: All 3
`mode: daemonset` & `mode: statefulset`: `OnDelete`, `RollingUpdate`

I've not implemented anything to reject the `Recreate` strategy for the other modes here.

Reply:
That's ok